src/client/sandbox/event/focus-blur.js

```diff
@@ -181,7 +148,36 @@ export default class FocusBlurSandbox extends SandboxBase {
 
         if (el[type]) {
             // NOTE: To guarantee that all focus/blur events are raised, we need to raise them manually.
-            this._ensureWindowScroll(el, preventScrolling, type, callback);
+            let windowScroll                           = null;
+            let elementNonScrollableParentsScrollState = null;
+
+            if (preventScrolling)
```

Comment:
May be we can extract in method storing scroll state `saveScrollStateIfNecessary` and for storing valuesuse` this`.
And after calling `focus` use method` restoreScrollStateIfNecessary`